### PR TITLE
Fix: camp creation validator logic

### PR DESCRIPTION
**Pull request by e-wai**
Add states to double check for non-negative, lower < upper, etc.

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)

## Implementation description
* 

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by twilkhoo — DISMISSED** [review]
LGTM, everything works on my end, just wondering if a camp's fee can ever be $0? Your implementation requires camp fees to be >=$1, so just curious about that.

**Review by twilkhoo — COMMENTED** [review]
One more question- do we want error validation for EDLP similar for start and end time? I.e. if early dropoff time is later than late pickup time, that should also be an error right?

File: frontend/src/components/pages/CampCreation/index.tsx [hunk 1]
```diff
@@ -59,18 +59,42 @@ const CampCreationPage = (): React.ReactElement => {
...
+  const startTimeBeforeEndTime = useMemo(() => {
+    try {
+      const [startHours, startMinutes] = startTime
+        .split(":")
+        .map((num) => parseInt(num, 10));
+      const [endHours, endMinutes] = endTime
+        .split(":")
+        .map((num) => parseInt(num, 10));
+
+      if (!(startHours && startMinutes && endHours && endMinutes)) {
```
Comment (twilkhoo) [inline frontend/src/components/pages/CampCreation/index.tsx:71]: Replace this if condition with:
```
if (Number.isNaN(startHours) || Number.isNaN(endHours)) {
  return true;
}
```
Right now, if you have times with 0, these will also pass the validation due to falsey value logic. Also, there's no need to check for startMinutes in the if condition, if either hours or minutes are not present, then hours will return NaN.

**Review by twilkhoo — APPROVED** [review]
LGTM

File: frontend/src/components/pages/CampCreation/index.tsx [hunk 2]
```diff
@@ -59,18 +59,76 @@ const CampCreationPage = (): React.ReactElement => {
...
+          .map((num) => parseInt(num, 10));
+        const [endHours, endMinutes] = end
+          .split(":")
+          .map((num) => parseInt(num, 10));
+
+        if (Number.isNaN(startHours) || Number.isNaN(endHours)) {
+          return true;
+        }
+
+        if (canBeEqual) {
```
Comment (twilkhoo) [inline frontend/src/components/pages/CampCreation/index.tsx:80]: Checked with Andy, EDLP is allowed to have the same times as the camp's start and end time, so I added this `canBeEqual` check.

**Comment by e-wai** [comment]
beauty

**Pull request merged**